File: test_nvidia_api.py
import requests
import json

def test_nvidia_api():
    print("🧪 Testing NVIDIA API")
    print("====================\n")
    
    base_url = "https://integrate.api.nvidia.com/v1"
    api_key = "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"
    
    headers = {
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json",
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36"
    }
    
    # Test 1: Get models endpoint
    print("Test 1: GET /models")
    try:
        response = requests.get(f"{base_url}/models", headers=headers, timeout=15)
        print(f"  Status: {response.status_code}")
        print(f"  Content-Type: {response.headers.get('content-type')}")
        
        if response.status_code == 200:
            try:
                data = response.json()
                print(f"  ✅ SUCCESS: Models retrieved")
                print(f"  Total models: {len(data.get('data', []))}")
                
                # Categorize models
                chat_models = []
                embedding_models = []
                other_models = []
                
                for model in data.get('data', []):
                    model_id = model.get('id', '')
                    if 'chat' in model_id.lower() or 'instruct' in model_id.lower() or 'conversational' in model_id.lower():
                        chat_models.append(model_id)
                    elif 'embed' in model_id.lower():
                        embedding_models.append(model_id)
                    else:
                        other_models.append(model_id)
                
                print(f"  Chat/Instruct models: {len(chat_models)}")
                print(f"  Embedding models: {len(embedding_models)}")
                print(f"  Other models: {len(other_models)}")
                
                print(f"\n  First 10 Chat/Instruct models: {chat_models[:10]}")
                print(f"  First 10 Embedding models: {embedding_models[:10]}")
                print(f"  First 10 Other models: {other_models[:10]}")
                
                print(f"\n  Complete Chat/Instruct model list ({len(chat_models)}):")
                for i, model in enumerate(chat_models, 1):
                    print(f"    {i}. {model}")
                
                return data
            except ValueError:
                print(f"  Response (text): {response.text[:300]}...")
                return response.text
        else:
            print(f"  ❌ FAILED: {response.text}")
            return None
            
    except Exception as e:
        print(f"  ❌ Error: {e}")
        return None

if __name__ == "__main__":
    result = test_nvidia_api()
    print('\n' + '='*50)
    if result:
        print('🎉 NVIDIA API Test: SUCCESS')
    else:
        print('❌ NVIDIA API Test: FAILED')

File: test_nvidia_api.js
import fetch from 'node-fetch';

// Test the NVIDIA API integration
async function testNVIDIAAPI() {
    console.log('Testing NVIDIA API integration...\n');
    
    const url = 'https://integrate.api.nvidia.com/v1/chat/completions';
    const apiKey = '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';
    
    const headers = {
        'Authorization': `Bearer ${apiKey}`,
        'Content-Type': 'application/json'
    };

    const payload = {
        "model": "moonshotai/kimi-k2-5",
        "messages": [
            {"role": "user", "content": "Hello"}
        ]
    };

    try {
        console.log('URL:', url);
        console.log('Headers:', headers);
        console.log('Payload:', JSON.stringify(payload, null, 2));
        
        const response = await fetch(url, {
            method: 'POST',
            headers: headers,
            body: JSON.stringify(payload)
        });

        console.log('\nStatus Code:', response.status);
        console.log('Status Text:', response.statusText);
        
        const responseBody = await response.text();
        console.log('Response Body:');
        console.log(responseBody);

        try {
            const jsonData = JSON.parse(responseBody);
            console.log('\nParsed JSON Response:');
            console.log(JSON.stringify(jsonData, null, 2));
        } catch (e) {
            console.log('Response is not valid JSON');
        }

        if (response.ok) {
            console.log('\n✅ NVIDIA API test successful!');
        } else {
            console.log(`\n❌ API returned error status: ${response.status}`);
        }

    } catch (error) {
        console.error('\n❌ Request failed:', error.message);
    }
}

// Run the test
await testNVIDIAAPI();